File: Golang Solutions/Course Schedule.go
func canFinish(numCourses int, prerequisites [][]int) bool {
    pre_count:=make([]int, numCourses)
    m:=make([][]int, numCourses)
    for _,x:= range prerequisites{
        pre_count[x[0]]++;
        m[x[1]]=append(m[x[1]],x[0]);
    }
    s:=[]int{}
    for i:=0;i<numCourses;i++{
        if pre_count[i]==0{
            s=append(s,i); 
        }          
    }
    var count int
    for count=0;len(s)!=0;count++{
        x:= s[0]
        s=s[1:]
        for _,next:=range m[x]{
            pre_count[next]--;
            if pre_count[next]==0{
                s=append(s,next)
            }       
        }
    }
    return count==numCourses;
   
}

File: Course Schedule.cpp
class Solution {
public:
    bool canFinish(int numCourses, vector<vector<int>>& prerequisites) {
        vector<int>pre_count(numCourses,0);
        vector<vector<int>>m(numCourses);
        for(auto x: prerequisites){
           pre_count[x[0]]++;
           m[x[1]].push_back(x[0]);
        }
        stack<int>s;
        for(int i=0;i<numCourses;i++)
        {
            if(pre_count[i]==0)
                s.push(i);
        }
        int count=0;
        while(!s.empty())
        {
            count++;
            int x= s.top();
            s.pop();
            for(auto next: m[x]){
                pre_count[next]--;
                if(pre_count[next]==0)
                    s.push(next); 
            }
        }
        return count==numCourses;
    }
};
